subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/DefaultJavaPluginExtension.java

```diff
@@ -46,4 +74,38 @@ public JavaVersion getTargetCompatibility() {
     public void setTargetCompatibility(JavaVersion value) {
         convention.setTargetCompatibility(value);
     }
+
+    @Override
+    public void registerFeature(String name, Action<? super FeatureSpec> configureAction) {
+        // todo: how to deal with group/name/version changes/set too late?
```

Comment:
That is a valid concern that needs addressing indeed.

Reply:
Fixed